lib/purchases_flutter.dart

```diff
@@ -555,6 +560,47 @@ class Purchases {
     return purchaseResult;
   }
 
+  static Future<PurchaseResult> purchase(
+      PurchaseParams purchaseParams,
+  ) async {
+    final purchasableItem = purchaseParams.purchasableItem;
+    final googleProductChangeInfo = purchaseParams.googleProductChangeInfo;
+    final googleIsPersonalizedPrice = purchaseParams.googleIsPersonalizedPrice;
+    final prorationMode = googleProductChangeInfo?.prorationMode?.value;
+    final signedDiscountTimestamp = purchaseParams.promotionalOffer?.timestamp.toString();
+    final presentedOfferingContext = purchaseParams.purchasableItem.presentedOfferingContext?.toJson();
+    final purchaseArgs = <String, dynamic>{
+      'googleOldProductIdentifier': googleProductChangeInfo?.oldProductIdentifier,
+      'googleProrationMode': prorationMode,
+      'googleIsPersonalizedPrice': googleIsPersonalizedPrice,
+      'signedDiscountTimestamp': signedDiscountTimestamp,
+      'presentedOfferingContext': presentedOfferingContext,
+    };
+    if (purchasableItem is Package) {
+      return await _invokeReturningPurchaseResult('purchasePackage', {
```

Comment:
This still uses the same native code to handle the purchases. I just changed the dart API and how we process these there. We could handle this logic in the native layer and just pass the entire purchaseParams... but that was more complex, and I think it would lead to extra duplication of logic in iOS/Android/web